src/context.ml

```diff
@@ -105,13 +106,39 @@ let sexp_of_t t =
     ; "findlib_path", list path (Findlib.path t.findlib)
     ; "arch_sixtyfour", bool t.arch_sixtyfour
     ; "natdynlink_supported", bool t.natdynlink_supported
+    ; "supports_shared_libraries", bool t.supports_shared_libraries
     ; "opam_vars", string_hashtbl string t.opam_var_cache
     ; "ocaml_config", Ocaml_config.sexp_of_t t.ocaml_config
     ; "which", string_hashtbl (option path) t.which_cache
     ]
 
 let compare a b = compare a.name b.name
 
+(* Parse the [`ocamlc -where`/makefile_config] file *)
+module Makefile_config = struct
```

Comment:
What do you think about putting this in its own module and making `t` abstract? This is a small addition, but it's part of a large module so it seems worth doing it to keep things manageable.

Reply:
Looking at this again, this code should clearly go in the ocaml_config library. I moved it there